### ISSUE TITLE
[wpr-go] When multiple requests match the same url, break tie using request headers

### ISSUE THREAD

**Issue by xunjieli** [catapult-project/catapult#3727]
In "load:tools:gmail" story, there are two serialized requests with the same URLs (GET mail.google.com/mail). Both result in 302s. The current wpr-go implementation will just pick the first one because the path matches exactly. This results in incorrect redirect of the login page. 
We should be redirected to https://mail.google.com/mail/u/0/.

We can break the tie by calculating a match ratio of request headers.

-------FIRST REQUEST---------------------------------
GET /mail/ HTTP/1.1
Host: mail.google.com
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.72 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8
Cookie: <a short cookie>

HTTP/2.0 302 Found
....
Location: https://accounts.google.com/ServiceLogin?service=mail&passive=true&rm=false&continue=https://mail.google.com/mail/&ss=1&scc=1&ltmpl=default&ltmplcache=2&emr=1&osid=1#

------SECOND REQUEST-------------------------
GET /mail/ HTTP/1.1
Host: mail.google.com
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.72 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8
Cookie: <a long cookie>

HTTP/2.0 302 Found
....
Location: https://mail.google.com/mail/u/0/

master bug #3629
/cc @tombergan @nedn

**Comment by tombergan** [comment]
Do the first and second request have different Cookies that you elided? The request headers look the same in both cases.

**Comment by tombergan** [comment]
Do the first and second request have different Cookies that you elided? The request headers look the same in both cases.

**Comment by nedn** [comment]
I remember the right way to fix this is to implement cache decay? @nyaxt

**Comment by xunjieli** [comment]
Yes, the first request has a short cookie and the second request has a long cookie.
Can I match the Cookie request header when selecting a request-response pair from the archive?
It seems that python WPR strips Cookie header all together though.

**Comment by tombergan** [comment]
Matching headers seems fine if you're OK with storing cookies in the archive. I think @nedn suggested ordering requests with the same URL? That is, if the record log has requests A, B, C for some URL (in that order), then during replay, we'd serve A, then B, then C, then maybe A again.

**Comment by nedn** [comment]
Correct @tombergan
